apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/commons/PreAdjClaimDaoTest.java

```diff
@@ -0,0 +1,165 @@
+package gov.cms.bfd.server.war.commons;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.codahale.metrics.MetricRegistry;
+import gov.cms.bfd.server.war.r4.providers.IPreAdjClaimResponseTypeV2;
+import gov.cms.bfd.server.war.r4.providers.IPreAdjClaimTypeV2;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.junit.Test;
+
+public class PreAdjClaimDaoTest {
+
+  @Test
+  public void shouldInvokeGetByIdForClaimType() {
+    String claimId = "123";
+
+    MetricRegistry registry = new MetricRegistry();
```

Comment:
Maybe include a check that the timer metric was updated?

Reply:
We could (though not for this specific test, as the target method doesn't invoke the `MetricRegistry`).  The only downside is that the `MetricRegistry` class is not easily mockable due to use of some private inner classes (`Timer.Context`), and it's recorded using a static invocation (also difficult to mock without bringing in a hefty testing library like PowerMock).  If there was ever an issue with the dependent classes, it could break a lot of tests making the cause a bit more ambiguous. 
 Might be worth looking at that as a test case for an Integration Test or at least something "above" a unit test.

Reply:
OK if the timer isn't being updated here you can disregard my comment.  WRT testing a metric though it's trivial (at least for Meters) to query the registry for the metric and check it's value.  For example for a Meter I did this:

````
  private void assertMeterReading(long expected, String meterName) {
    long actual = appMetrics.meter(meterName).getCount();
    assertEquals("Meter " + meterName, expected, actual);
  }
````